definitions/pipelines/immuno.cwl

```diff
@@ -0,0 +1,467 @@
+        in:
+            reference: reference
+            bams: normal_bams
+            readgroups: normal_readgroups
+            mills: mills
+            known_indels: known_indels
+            dbsnp_vcf: dbsnp_vcf
+            bqsr_intervals: bqsr_intervals
+            bait_intervals: bait_intervals
+            target_intervals: target_intervals
+            per_base_intervals: per_base_intervals
+            per_target_intervals: per_target_intervals
+            summary_intervals: summary_intervals
+            omni_vcf: omni_vcf
+            picard_metric_accumulation_level: picard_metric_accumulation_level
+            emit_reference_confidence: emit_reference_confidence
+            gvcf_gq_bands: gvcf_gq_bands
+            intervals: gatk_haplotypecaller_intervals
+            vep_cache_dir: vep_cache_dir
+            synonyms_file: synonyms_file
+            annotate_coding_only: annotate_coding_only
+            custom_gnomad_vcf: custom_gnomad_vcf
+            qc_minimum_mapping_quality: qc_minimum_mapping_quality
+            qc_minimum_base_quality: qc_minimum_base_quality
+            custom_clinvar_vcf: custom_clinvar_vcf
+            optitype_name: optitype_name
+            vep_assembly: vep_assembly
+        out:
+            [cram,mark_duplicates_metrics,insert_size_metrics,insert_size_histogram,alignment_summary_metrics,hs_metrics,per_target_coverage_metrics,per_target_hs_metrics,per_base_coverage_metrics,per_base_hs_metrics,summary_hs_metrics,flagstats,verify_bam_id_metrics,verify_bam_id_depth,gvcf,final_vcf,coding_vcf,limited_vcf,vep_summary,optitype_tsv,optitype_plot]
+    phase_vcf:
+        run: ../subworkflows/phase_vcf.cwl
+        in:
+            somatic_vcf: somatic/final_vcf
+            germline_vcf: germline/final_vcf
+            reference: reference
+            reference_dict: reference_dict
+            bam: somatic/tumor_cram
+        out:
+            [phased_vcf]
+    extract_alleles:
+        run: ../tools/extract_hla_alleles.cwl
+        in:
+            allele_file: germline/optitype_tsv
+        out:
+            [allele_string]
+    pvacseq:
+        run: ../subworkflows/pvacseq.cwl
+        in:
+            detect_variants_vcf: somatic/final_vcf
+            sample_name: 
+                default: 'TUMOR'
+            normal_sample_name: 
+                default: 'NORMAL'
+            rnaseq_bam: rnaseq/final_bam
+            reference_fasta: reference
+            readcount_minimum_base_quality: readcount_minimum_base_quality #should this be different from qc_minimum_base_quality?
+            readcount_minimum_mapping_quality: readcount_minimum_mapping_quality #should this be different from qc_minimum_mapping_quality
+            gene_expression_file: rnaseq/stringtie_gene_expression_tsv #not sure if this connection is correct
+            transcript_expression_file: rnaseq/transcript_abundance_tsv #not sure if this connection is correct
+            alleles: extract_alleles/allele_string
```

Comment:
Since we often do clinical HLA-typing, is there a reasonable way to allow the user to provide an HLA string, that overrides whatever Optitype finds?

Reply:
Since there are still no conditionals in CWL, I'm not sure if there is a reasonably simple way to do this. Suggestions from anyone? Maybe provide `auto_alleles` and `manual_alleles` as inputs to the tool, where manual is optional and overrides auto if present?

Reply:
Following discussion, this will be added in v2